### PR TITLE
Create alignment ADR

### PR DESCRIPTION
**Pull request by BlueReZZ**
This has been looked over and tweaked by a few people in Product and Tech so no ready to be recorded here

### CODE REVIEW

**Comment by Gman0909** [comment]
Looks good to me

File: doc/adr/0004-align-technology-practice-process.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+The process is to explore every reasonable opportunity for a strong alignment, and to pursue it in practice when it becomes apparent that its implementation will not be detrimental to the web apps’ end user experience. Therefore we are not seeking alignment for alignment’s sake, nor are we ignoring it for the sake of the web apps’ end users.
+
+There will always be a balance between user experience, developer experience and publishing staff experience. With software-as-a-service or proprietary software products, the end user experience can always be prioritised over the others, but the community-driven, open source nature of the Libero suite means we have to consider all three. Below is the order in which we should ask ourselves the questions during discovery of an area that could be aligned:
+
+1. Why should we align in this area? Why should we do something different? What is the value?
+2. Can we strongly align in this area across products?
+3. What are the consequences for the web site / app end users or publishing staff experience?
```
Comment (diversemix) [inline doc/adr/0004-align-technology-practice-process.md:31]: How about "What are the consequences for the above personnas?" ?

Comment (diversemix) [reply doc/adr/0004-align-technology-practice-process.md:31]: :thinking: ... this is probably best as a grid - personnas vs degree of alignment? Where each cell details the level of how they are affected.

**Review by diversemix — APPROVED** [review]

File: doc/adr/0004-align-technology-practice-process.md (hunk 1 above)
Comment (BlueReZZ) [reply doc/adr/0004-align-technology-practice-process.md:31]: The intent is to ensure that the developer persona is considered differently to the others so I think the separation as it is makes sense. I like the idea of a table but want to detail a set of questions to spark a discussion and not prescribe a form to fill in - no harm in people recording the discovery in that way though.

File: doc/adr/0004-align-technology-practice-process.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+- Service providers who want to configure and deploy instances of the products
+
+### Decision
+
+The approach is to align in a number of well-defined areas and record this in ADRs. However, strong alignment is not a requirement where it is detrimental to the way the product solves the problems for its given use cases or where it creates undesirable coupling between teams (resulting in synchronisation overheads and reduced autonomy/empowerment). For example, using the same framework for Publisher and Reviewer on the front-end would mean that either single-page performance or interactivity, respectively, would have to suffer, so a better choice would be weaker alignment in that area, as Publisher requires the first and Reviewer depends on the latter.
+
+The process is to explore every reasonable opportunity for a strong alignment, and to pursue it in practice when it becomes apparent that its implementation will not be detrimental to the web apps’ end user experience. Therefore we are not seeking alignment for alignment’s sake, nor are we ignoring it for the sake of the web apps’ end users.
```
Comment (giorgiosironi) [inline doc/adr/0004-align-technology-practice-process.md:25]: this paragraph seems to suggest always to explore a strong alignment first, and pursue if it's not detrimental; but the list of questions say the first thought is consider whether there is positive value in the alignment?

**Review by giorgiosironi — APPROVED** [review]

File: doc/adr/0004-align-technology-practice-process.md (hunk 2 above)
Comment (BlueReZZ) [reply doc/adr/0004-align-technology-practice-process.md:25]: Good observation. The intent is to explore strong alignment but then ensure we have considered and record the value during the discovery - hence the need for the question. Hopefully, most of the time, the value will be easily obvious ;) The inclusion of the question on the potential diversion prompts us to consider this from either starting point. Following all the questions should get us to the result we want.

**Pull request merged**